### PR TITLE
Csnizik/205790 timeline right slot

### PR DESCRIPTION
**Pull request by csnizik**
### Summary: adds a content slot to the right of the link title. 
**UPDATED** https://www.figma.com/file/hDfTdVzAAuf2DDb3hVSnFM/Notebooks%3A-Feedback-Overview?node-id=366%3A26162

Changed story to use dot icons (but any ReactNode could go there)

### Test Plan:
- yarn lint passes
- yarn test passes
- yarn types passes
- yarn start works

### CODE REVIEW

File: src/components/TimelineNavPanel/TimelineNavPanel.tsx [hunk 1]
```diff
@@ -57,6 +61,7 @@ export const TimelineNavPanel = ({
   className,
   completed,
   id,
+  right,
```
Comment (dierat) [inline src/components/TimelineNavPanel/TimelineNavPanel.tsx:64]: Where is this being used?

File: src/components/TimelineNav/TimelineNav.tsx [hunk 2]
```diff
@@ -359,6 +360,9 @@ export const TimelineNav = ({
           {timelineNavItems().map((tab: TimelineNavItem, i: number) => {
             const isActive = activeIndexState === i;
             const itemVariant = variant && tab.props.variant;
+            {
+              console.log({ tab: tab });
+            }
```
Comment (dierat) [inline src/components/TimelineNav/TimelineNav.tsx:365]: Did you mean to leave this in?

File: src/components/TimelineNav/TimelineNav.tsx [hunk 3]
```diff
@@ -109,6 +109,7 @@ export const TimelineNav = ({
   overflow,
   onChange,
   required,
+  right,
```
Comment (dierat) [inline src/components/TimelineNav/TimelineNav.tsx:112]: Where is this being used?

File: src/components/TimelineNav/TimelineNav.tsx [hunk 4]
```diff
@@ -48,44 +52,40 @@ export interface Props {
...
-  items?: Array<any>;
+  overflow?: 'inverted';
   /**
    * Indicates that field is required for form to be successfully submitted
    */
   required?: boolean;
+  /**
+   * Right slot: content that will be positioned to the right of title
+   */
+  right?: ReactNode;
```
Comment (dierat) [inline src/components/TimelineNav/TimelineNav.tsx:66]: I'm wondering if there is a more specific prop name we could use here, like `rightElement` or something of that nature.

File: src/components/TimelineNav/TimelineNav.stories.tsx [hunk 5]
```diff
@@ -28,7 +29,11 @@ export default {
             </p>
           </Text>
         </TimelineNavPanel>
-        <TimelineNavPanel title="TimelineNavPanel 1" variant="number">
+        <TimelineNavPanel
+          right={<Tag hasOutline text="Feedback" variant="brand" />}
```
Comment (dierat) [inline src/components/TimelineNav/TimelineNav.stories.tsx:33]: Sounds like design landed on using a dot here: [figma spec](https://www.figma.com/file/hDfTdVzAAuf2DDb3hVSnFM/Notebooks%3A-Feedback-Overview?node-id=366%3A26162)

File: src/components/TimelineNav/TimelineNav.tsx [hunk 6]
```diff
@@ -31,6 +31,10 @@ export interface Props {
    * The aria-labelledby attribute creates a relationship between the tab list and the tab panels
    */
   'aria-labelledby'?: string;
+  /**
+   * Passed down to initially set the activeIndex state
+   */
+  activeIndex?: number;
```
Comment (dierat) [inline src/components/TimelineNav/TimelineNav.tsx:37]: Thanks for alphabetizing these! ❤️

File: src/components/TimelineNav/TimelineNav.tsx (hunk 2 above)
Comment (csnizik) [reply src/components/TimelineNav/TimelineNav.tsx:365]: Oopsie lol

File: src/components/TimelineNavPanel/TimelineNavPanel.tsx (hunk 1 above)
Comment (csnizik) [reply src/components/TimelineNavPanel/TimelineNavPanel.tsx:64]: Looks like all of `<TimelineNavPanel>`'s props get passed to `<TimelineNav>`. In `<TimelineNav>` the panels, props and children get collected by timelineNavItems(), which gets iterated over by a .map() method inside the return; the props then get applied to each <li>.

File: src/components/TimelineNav/TimelineNav.tsx (hunk 3 above)
Comment (csnizik) [reply src/components/TimelineNav/TimelineNav.tsx:111]: See above.

File: src/components/TimelineNav/TimelineNav.tsx (hunk 4 above)
Comment (csnizik) [reply src/components/TimelineNav/TimelineNav.tsx:66]: Good catch. The `<PageHeader>` component uses the same prop and calls it `titleAfter` so I changed this to match.

File: src/components/TimelineNavPanel/TimelineNavPanel.tsx (hunk 1 above)
Comment (dierat) [reply src/components/TimelineNavPanel/TimelineNavPanel.tsx]: Could you add a quick comment, here and on the other one?

File: src/components/TimelineNav/TimelineNav.tsx (hunk 4 above)
Comment (dierat) [reply src/components/TimelineNav/TimelineNav.tsx]: Awesome, thanks!

**Review by ifrost1 — APPROVED** [review]

**Pull request merged**